### PR TITLE
Add Doc-Deploy ClickHouse

### PR DESCRIPTION
**Pull request by Felixnoo**
Signed-off-by: Felixnoo <felixliu@yunify.com>

### CODE REVIEW

File: content/en/docs/application-store/external-apps/deploy-clickhouse.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+- You need to create a workspace, a project, and two user accounts (`ws-admin` and `project-regular`) for this tutorial. The account `ws-admin` must be granted the role of `workspace-admin` in the workspace, and the account `project-regular` must be invited to the project with the role of `operator`. This tutorial uses `demo-workspace` and `demo-project` for demonstration. If they are not ready, refer to [Create Workspaces, Projects, Accounts and Roles](../../../quick-start/create-workspace-and-project/).
+
+## Hands-on Lab
+
+### Step 1: Deploy ClickHouse Operator
+
+1. Log in to the KubeSphere Web console as `admin`, and use **Kubectl** from the **Toolbox** in the bottom right corner to run the following command to install ClickHouse Operator.
+
+   ```bash
+   kubectl apply -f https://raw.githubusercontent.com/TCeason/clickhouse-helm-chart/main/clickhouse-operator-install.yml
```
Comment (FeynmanZhou) [inline content/en/docs/application-store/external-apps/deploy-clickhouse.md:28]: We should not use a personal GitHub account in the documentation.

File: content/en/docs/application-store/external-apps/deploy-clickhouse.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+title: "Deploy ClickHouse Operator and a ClickHouse Cluster on KubeSphere"
+keywords: 'KubeSphere, Kubernetes, ClickHouse, ClickHouse Operator, ClickHouse Cluster'
+description: 'Learn how to deploy ClickHouse Operator and a ClickHouse Cluster on KubeSphere.'
+linkTitle: "Deploy ClickHouse Operator and a ClickHouse Cluster on KubeSphere"
+weight: 14340
+---
+
+[ClickHouse](https://clickhouse.tech/) is a column-oriented database management system (DBMS) for online analytical processing of queries (OLAP). ClickHouse on QingCloud is a ClickHouse cluster application deeply customized with the following features.
+
+- Automated cluster management. ClickHouse on QingCloud manages cluster status via Paxos protocols. As you run ClickHouse on QingCloud and manage it like a single-node ClickHouse cluster, you will be free from complicated cluster management operations.
```
Comment (Sherlock113) [inline content/en/docs/application-store/external-apps/deploy-clickhouse.md:11]: ClickHouse on QingCloud manages cluster status via Paxos protocols.
->
ClickHouse on QingCloud manages cluster status via the Paxos protocol.

File: content/en/docs/application-store/external-apps/deploy-clickhouse.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+- Data redistribution in clusters. Apart from automated management of the cluster, you can also perform cluster-level data operations, including `SYSTEM RESHARDING`, `SYSTEM RESHARDING DATABASE [database_name]`, and `SYSTEM RESHARDING TABLE [database_name].[table_name]`. Since you can redistribute your cluster data based on you needs, data query performance will be evidently optimized with less scenarios like inappropriate data distributions.
```
Comment (Sherlock113) [inline content/en/docs/application-store/external-apps/deploy-clickhouse.md:12]: data query performance will be evidently optimized with less scenarios like inappropriate data distributions.
->
data query performance will be greatly optimized.

File: content/en/docs/application-store/external-apps/deploy-clickhouse.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+3. On the new page that appears, select **clickhouse** from the drop-down list and then click **clickhouse-cluster**.
+
+   ![clickhouse-cluster](/images/docs/appstore/external-apps/deploy-clickhouse/clickhouse-cluster.png)
+
+4. On the **Chart Files** tab, you can view the configuration and download the `values.yaml` file. Click **Deploy** to continue.
+
+   ![chart-tab](/images/docs/appstore/external-apps/deploy-clickhouse/chart-tab.png)
+
+5. On the **Basic Info** page, confirm the app name, app version, and deployment location. Click **Next** to continue.
```
Comment (Sherlock113) [inline content/en/docs/application-store/external-apps/deploy-clickhouse.md:109]: On the **Basic Info** page
->
On the **Basic Information** page

**Comment by Sherlock113** [comment]
The installation link of ClickHouse Operator needs to be changed, about which I already talked to our database doc colleague. Hold it for now.

File: content/en/docs/application-store/external-apps/deploy-clickhouse.md (hunk 1 above)
Comment (Sherlock113) [reply content/en/docs/application-store/external-apps/deploy-clickhouse.md:29]: @Felixnoo Use this link https://raw.githubusercontent.com/radondb/radondb-clickhouse-kubernetes/master/clickhouse-operator-install.yml

File: content/en/docs/application-store/external-apps/deploy-clickhouse.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+   replicaset.apps/clickhouse-operator-6b8494c8f   1         1         1       6m34s
+   ```
+
+### Step 2: Add an app repository
+
+1. Log out of KubeSphere and log back in as `ws-admin`. In **demo-workspace**, go to **App Repositories** under **App Management**, and then click **Add**.
+
+   ![add-repo](/images/docs/appstore/external-apps/deploy-clickhouse/add-repo.png)
+
+2. In the dialog that appears, enter `clickhouse` for the app repository name and `https://tceason.github.io/clickhouse-helm-chart/` for the repository URL. Click **Validate** to verify the URL and you will see a green check mark next to the URL if it is available. Click **OK** to continue.
```
Comment (Sherlock113) [inline content/en/docs/application-store/external-apps/deploy-clickhouse.md:83]: Use this repository: https://radondb.github.io/radondb-clickhouse-kubernetes/

File: content/en/docs/application-store/external-apps/deploy-clickhouse.md [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+
+- Automated cluster management. ClickHouse on QingCloud manages cluster status via the Paxos protocol. As you run ClickHouse on QingCloud and manage it like a single-node ClickHouse cluster, you will be free from complicated cluster management operations.
+- Data redistribution in clusters. Apart from automated management of the cluster, you can also perform cluster-level data operations, including `SYSTEM RESHARDING`, `SYSTEM RESHARDING DATABASE [database_name]`, and `SYSTEM RESHARDING TABLE [database_name].[table_name]`. Since you can redistribute your cluster data based on you needs, data query performance will be greatly optimized.
+- Excellent performance and less cost. ClickHouse on QingCloud adjusts certain key parameters and carries the excellent performance of ClickHouse clusters over into both single-node and multi-node cluster environments with various disk types supported. You can choose the disk type based on your needs and experience the excellent performance with no need to adjust any parameters.
```
Comment (Sherlock113) [inline content/en/docs/application-store/external-apps/deploy-clickhouse.md:13]: I am actually thinking if these paragraphs are appropriate as typically we do not put this kind of content in a guide. Even if they are included, there should not be too much of them.

Comment (Felixnoo) [reply content/en/docs/application-store/external-apps/deploy-clickhouse.md:13]: If that's the case, I think we can write down something like `ClickHouse on QingCloud is a ClickHouse cluster application deeply customized with features including automated cluster management, data redistribution in clusters, and excellent performance with less cost.`

**Comment by Felixnoo** [comment]
Issues to be confirmed. Hold this PR for now.

**Comment by Felixnoo** [comment]
@Sherlock113 Ready for review.

File: content/en/docs/application-store/external-apps/deploy-clickhouse.md (hunk 6 above)
Comment (Sherlock113) [reply content/en/docs/application-store/external-apps/deploy-clickhouse.md]: For the introduction part, refer to https://github.com/kubesphere/website/pull/1505.

File: content/en/docs/application-store/external-apps/deploy-clickhouse.md [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+   NAME                                  READY   UP-TO-DATE   AVAILABLE   AGE
+   deployment.apps/clickhouse-operator   1/1     1            1           6m34s
+   
+   NAME                                            DESIRED   CURRENT   READY   AGE
+   replicaset.apps/clickhouse-operator-6b8494c8f   1         1         1       6m34s
+   ```
+
+### Step 2: Add an app repository
+
+1. Log out of KubeSphere and log back in as `ws-admin`. In **demo-workspace**, go to **App Repositories** under **App Management**, and then click **Add**.
```
Comment (Sherlock113) [inline content/en/docs/application-store/external-apps/deploy-clickhouse.md:75]: **demo-workspace**
->
`demo-workspace`

File: content/en/docs/application-store/external-apps/deploy-clickhouse.md [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+
+   ![add-clickhouse](/images/docs/appstore/external-apps/deploy-clickhouse/add-clickhouse.png)
+
+3. Your repository displays in the list after successfully imported to KubeSphere.
+
+   ![repo-added](/images/docs/appstore/external-apps/deploy-clickhouse/repo-added.png)
+
+### Step 3: Deploy a ClickHouse Cluster
+
+1. Log out of KubeSphere and log back in as `project-regular`. In **demo-project**, go to **Apps** under **Application Workloads** and click **Deploy New App**.
```
Comment (Sherlock113) [inline content/en/docs/application-store/external-apps/deploy-clickhouse.md:89]: **demo-project**
->
`demo-project`

File: content/en/docs/application-store/external-apps/deploy-clickhouse.md [hunk 9]
```diff
@@ -0,0 +1,174 @@
+---
+title: "Deploy ClickHouse Operator and a ClickHouse Cluster on KubeSphere"
+keywords: 'KubeSphere, Kubernetes, ClickHouse, ClickHouse Operator, ClickHouse Cluster'
+description: 'Learn how to deploy ClickHouse Operator and a ClickHouse Cluster on KubeSphere.'
+linkTitle: "Deploy ClickHouse Operator and a ClickHouse Cluster on KubeSphere"
+weight: 14340
+---
+
+[ClickHouse](https://clickhouse.tech/) is a column-oriented database management system (DBMS) for online analytical processing of queries (OLAP). ClickHouse on QingCloud is a ClickHouse cluster application deeply customized with features including automated cluster management, data redistribution in clusters, and excellent performance with less cost.
```
Comment (Sherlock113) [inline content/en/docs/application-store/external-apps/deploy-clickhouse.md:9]: with features including
->
featuring

**Pull request merged**